File: 2024/Desenvolvimento_para_Dispositivos_Moveis/10- Outubro/4/flutter_application_prova_pratica_api/lib/animal.dart
class Animal{
  String _nomeC = "";
  String _nomeP = "";
  String _riscoExt = "";
  String _habito = "";

  Animal();

  Animal.fromJson(Map <String, dynamic> json):
    _nomeC = json["nomeCientifico"],
    _nomeP = json["nomePopular"],
    _riscoExt = json["riscoExtincao"],
    _habito = json["habito"]
  ;

  String get nomeC => this._nomeC;

  set nomeC( value) => this._nomeC = value;

  String get nomeP => this._nomeP;

  set nomeP( value) => this._nomeP = value;

  String get riscoExt => this._riscoExt;

  set riscoExt( value) => this._riscoExt = value;

  String get habito => this._habito;

  set habito( value) => this._habito = value;
}

File: 2024/Desenvolvimento_para_Dispositivos_Moveis/10- Outubro/4/ProvaPraticaAPI/src/main/java/com/example/ProvaPraticaAPI/model/Animal.java
package com.example.ProvaPraticaAPI.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table (name = "Animal")
public class Animal {

    @Id
    private String nomeCientifico;

    @Column
    private String nomePopular;

    @Column
    private String riscoExtincao;

    @Column
    private String habito;

    public String getNomeCientifico() {
        return nomeCientifico;
    }

    public void setNomeCientifico(String nomeCientifico) {
        this.nomeCientifico = nomeCientifico;
    }

    public String getNomePopular() {
        return nomePopular;
    }

    public void setNomePopular(String nomePopular) {
        this.nomePopular = nomePopular;
    }

    public String getRiscoExtincao() {
        return riscoExtincao;
    }

    public void setRiscoExtincao(String riscoExtincao) {
        this.riscoExtincao = riscoExtincao;
    }

    public String getHabito() {
        return habito;
    }

    public void setHabito(String habito) {
        this.habito = habito;
    }
}
